doc/howto/subcommand_argument_parser/subcommand_arg_parse.cpp

```diff
@@ -101,9 +100,12 @@ int main(int argc, char const ** argv)
     std::cout << "Proceed to sub parser.\n";
 
     if (sub_parser.info.app_name == std::string_view{"mygit-pull"})
-        run_git_pull(sub_parser);
+        return run_git_pull(sub_parser);
     else if (sub_parser.info.app_name == std::string_view{"mygit-push"})
-        run_git_push(sub_parser);
+        return run_git_push(sub_parser);
     else
-        throw std::logic_error{"I do not know sub parser " + sub_parser.info.app_name};
+        std::cout << "Did you catch the wrong names? The subparser is called: " << sub_parser.info.app_name;
+        // Note: If you set up your if branches correctly, you should never reach this else block and you can omit it.
```

Comment:
Why would we intentionally do it wrong? :)

Maybe you can rephrase it to state more clearly that you only should reach this if you add a subparser and forget to handle it down here (otherwise it would've thrown in `parse()`)

Reply:
Good point, I tried :) Take a look and make a suggestion if it is still confusing.